rebasehelper/specfile.py

```diff
@@ -102,6 +103,82 @@ def get_strip(self):
         return self.strip
 
 
+class SpecContent(object):
+    """Class representing content of a SPEC file."""
+
+    SECTION_HEADERS = [
+        '%package',
+        '%prep',
+        '%build',
+        '%install',
+        '%check',
+        '%clean',
+        '%prerun',
+        '%postrun',
+        '%pretrans',
+        '%posttrans',
+        '%pre',
+        '%post',
+        '%files',
+        '%changelog',
+        '%description',
+        '%triggerpostun',
+        '%triggerprein',
+        '%triggerun',
+        '%triggerin',
+        '%trigger',
+        '%verifyscript',
+        '%sepolicy',
+        '%filetriggerin',
+        '%filetrigger',
+        '%filetriggerun',
+        '%filetriggerpostun',
+        '%transfiletriggerin',
+        '%transfiletrigger',
+        '%transfiletriggerun',
+        '%transfiletriggerpostun',
+    ]
+
+    def __init__(self, content):
+        self.sections = CaseInsensitiveDict()
+        self._split_sections(content)
+
+    def __str__(self):
+        """Join SPEC file sections back together."""
+        content = []
+        for header, section in six.iteritems(self.sections):
+            if header != '%package':
+                content.append(header + '\n')
+            for line in section:
+                content.append(line + '\n')
+        return ''.join(content)
+
+    def _split_sections(self, content):
+        """Splits content of a SPEC file into sections.
+
+        Args:
+            content (str): Content of the SPEC file
+
+        """
+        lines = content.splitlines()
+        section_headers_re = [re.compile(r'^{0}.*'.format(re.escape(x)), re.I) for x in self.SECTION_HEADERS]
+
+        section_beginnings = sb = []
+        for i, line in enumerate(lines):
+            if line.startswith('%'):
+                for header in section_headers_re:
+                    if header.match(line):
+                        section_beginnings.append(i)
+        section_beginnings.append(None)
+
+        self.sections['%package'] = lines[:section_beginnings[0]]
+
+        for i in range(len(section_beginnings) - 1):
+            start = sb[i] + 1
+            end = sb[i + 1]
+            self.sections[lines[sb[i]]] = lines[start:end]
```

Comment:
I don't think shortening `section_beginnings` is necessary now, and you could also use lines[start - 1] here.